RBAC Reference for Enterprise (#473)
astronomer/docs: enterprise/v0.25/06_manage-astronomer/02_manage-platform-users.md

- `system.invites.get`: View all pending user invites
- `system.invite.get`: View information for any pending user invite
- `system.monitoring.get`: View [Grafana](/docs/enterprise/v0.25/monitor/grafana-metrics) and [Kibana](/docs/enterprise/v0.25/monitor/kibana-logging) metrics for any Deployment
- `system.serviceAccounts.get`: View [service accounts](/docs/enterprise/v0.25/deploy/ci-cd#step-1-create-a-service-account) for any Deployment or Workspace
- `system.updates.get`: View the newest platform release version number
- `system.users.get`: View any user on the platform
- `system.workspace.get`: View information for any Workspace

### System Editor

The System Editor has all of the same default permissions as the System Viewer, plus:

- `system.deployment.variables.update`: Modify [environment variables](/docs/cloud/stable/deploy/environment-variables) for any Deployment
- `system.iam.update`: Modify [IAM](/docs/enterprise/v0.25/customize-airflow/integrate-iam) roles for any Deployment
- `system.serviceAccounts.update`: Modify [service accounts](/docs/enterprise/v0.25/deploy/ci-cd#step-1-create-a-service-account) for any Workspace or Deployment

> - `deployment.airflow.user`: Airflow [user permissions](https://airflow.apache.org/docs/apache-airflow/stable/security/access-control.html#user) for all Deployments

Reviewer: ```suggestion
- `deployment.airflow.user`: Modify Airflow [user permissions](https://airflow.apache.org/docs/apache-airflow/stable/security/access-control.html#user) for all Deployments
```

Verb here?

Author: You don't modify the permissions, you just have the permissions in that link. Adding another sentence to clarify what those permissions are.

> - `system.registryBaseImages.push`: Modify base layer Docker images for Airflow

Reviewer: Tbh not sure what this means, might also be good to ask 🙈

Author: @aliotta tagging you for your expertise again! Do we have any more information/context that we could share about this permission?

### System Admin

The System Admin has all of the same default permissions as the System Viewer and System Editor for a given cluster, plus:

- `system.deployments.create`: Create a Deployment on any Workspace
- `system.deployments.update`: Modify any Deployment
- `system.deployments.delete`: Delete any Deployment

> - `system.deployments.logs`: View logging and metrics for any Deployment

Reviewer: Double check this? Sounds like these are Deployment logs (tab in Astro UI) - can system viewers not access them?

Author: That is what it says here: https://github.com/astronomer/houston-api/blob/c50a0d602c4b7815af34ce7c64c6a07673740503/config/default.yaml#L314

Not sure why, but my guess is that we do this based on a "least access" model where system viewers only need Kibana and Grafana and don't need specific Deployment logs/ metrics.

> - `system.user.invite`: Invite any user

Reviewer: ```suggestion
- `system.user.invite`: Invite a user
```

- `system.user.delete`: Delete any user
- `system.user.verifyEmail`: Bypass email verification for any user

> - `system.workspace.addCustomerId`: Update billing information for any Workspace

Reviewer: Not applicable for Enterprise - can delete (`addCustomerId`)

If we feel like Enterprise customers have access to our GitHub repos and will see these permissions listed somewhere, we could add a note that references this one and the suspend workspace one and just say they're operational permissions for Astro Cloud 🤷‍♀️

Not sure on the visibility here - will defer but just an idea

- `system.workspace.delete`: Delete any Workspace

> - `system.airflow.admin`: Airflow admin permissions, including permission to configure:

Reviewer: ```suggestion
- `system.airflow.admin`: Airflow admin permissions on any Deployment, including permission to configure:
```

    - Pools
    - Configuration
    - Users
    - Connections
    - Variables
    - XComs

### Workspace Viewer

The Workspace Viewer has the following default permissions for a given Workspace:

- `workspace.config.get`: View the Workspace

> - `workspace.deployments.get`: View any Deployment

Reviewer: ```suggestion
- `workspace.deployments.get`: View all Deployments
```

- `workspace.serviceAccounts.get`: View any Deployment or Workspace-level [service account](/docs/enterprise/v0.25/deploy/ci-cd#step-1-create-a-service-account)

> - `workspace.users.get`: View all user information

Reviewer: ```suggestion
- `workspace.users.get`: View information for all users with access to the Workspace
```

### Workspace Editor

For a given Workspace, the Workspace Editor has all of the same default permissions as the Workspace Viewer, plus:

> - `workspace.config.update`: Modify the Workspace

Reviewer: ```suggestion
- `workspace.config.update`: Modify the Workspace, including Workspace Name, Description, and user access
```

- `workspace.deployments.create`: Create a Deployment in the Workspace
- `workspace.serviceAccounts.create`: Create a Workspace-level service account
- `workspace.serviceAccounts.update`: Modify a Workspace-level service account
- `workspace.serviceAccounts.delete`: Delete a Workspace-level service account

### Workspace Admin

For a given Workspace, the Workspace Editor has all of the same default permissions as the Workspace Viewer and Workspace Editor, plus:

> - `workspace.billing.update:` Update billing information for the Workspace

Reviewer: ```suggestion
- `workspace.billing.update:` Update billing information for the Workspace
```

Cloud only

- `workspace.invites.get`: View pending user invites for the Workspace
- `workspace.config.delete`: Delete the Workspace
- `workspace.iam.update`: Update [IAM](/docs/enterprise/v0.25/customize-airflow/integrate-iam) for the Workspace

### Deployment Viewer

For a given Deployment, a Deployment Viewer has the following permissions:

- `deployment.airflow.get`: View the Airflow UI
- `deployment.config.get`: View the Deployment's settings
- `deployment.logs.get`: View the Deployment's logs
- `deployment.images.pull`: Access the Deployment's running Docker image
- `deployment.metrics.get`: View the Deployment's **Metrics** tab in the Astronomer UI
- `deployment.serviceAccounts.get`: View any [service account](/docs/enterprise/v0.25/deploy/ci-cd#step-1-create-a-service-account) for the Deployment
- `deployment.variables.get`: View the Deployment's [environment variables](/docs/cloud/stable/deploy/environment-variables)

> - `deployment.users.get`: View the Deployment's users

Reviewer: ```suggestion
- `deployment.users.get`: View the list of users with access to the Deployment
```

> - `deployment.users.get`: View the Deployment's users

Reviewer: In our Gen1 user docs, we have a note about what happens when a Deployment viewer tries to modify/access something they can't:

> Deployment Viewers cannot deploy to, modify, or delete anything within an Airflow Deployment. Additionally, they cannot create or use Service Accounts to do so. Attempts to modify a Deployment in any way will result in a 403 and an Access is Denied message.

Might be a good idea to add here? https://www.astronomer.io/docs/cloud/stable/manage-astronomer/workspace-permissions

- Navigate to `grafana.BASEDOMAIN`
- Navigate to `kibana.BASEDOMAIN`
- Access the 'System Admin' tab from the top left menu of the Astronomer UI

## Reference: Platform Permissions

> Use this section to see the specific default permissions for each user type on Astronomer.

Author: ```suggestion
This section contains the specific default permissions for each user type on Astronomer.
```

### System Viewer

The System Viewer has the following default roles:

- `system.airflow.get`: View the Airflow UI for any Deployment
- `system.deployment.variables.get`: View environment variables for any Deployment
- `system.deployments.get`: View settings and configurations for any Deployment
- `system.invites.get`: View all pending user invites
- `system.invite.get`: View information for any pending user invite
- `system.monitoring.get`: View the Airflow UI for any Deployment
- `system.serviceAccounts.get`: See service accounts for any Deployment or Workspace

> - `system.updates.get`: [?]

Author: Values with `[?]` are unknown, reviewers should add the values if they know them

Reviewer 2: View the newest platform release version number
